roboteq_diff_driver/src/driver.cpp

```diff
@@ -781,67 +356,26 @@ void MainNode::odom_publish()
   float dt = (float)DELTAT(nowtime,odom_last_time) / 1000.0;
   odom_last_time = nowtime;
 
-#ifdef _ODOM_DEBUG
-/*
-ROS_DEBUG("right: ");
-ROS_DEBUG(odom_encoder_right);
-ROS_DEBUG(" left: ");
-ROS_DEBUG(odom_encoder_left);
-ROS_DEBUG(" dt: ");
-ROS_DEBUG(dt);
-ROS_DEBUG("");
-*/
-#endif
-
   // Based on the pre-existing math for computing `linear` and `angular`, I believe that odom_encoder_left if the number of encoder clicks *since the last message*
   // Thus the wheel rate is  2*pi * (clicks_since_last_time / clicks_per_rev) / dt
-  float wheel_rate_left = (static_cast<float>(odom_encoder_left) / static_cast<float>(encoder_cpr)) * 2*M_PI / dt;
-  float wheel_rate_right = (static_cast<float>(odom_encoder_right) / static_cast<float>(encoder_cpr)) * 2*M_PI / dt;
-
 
   // determine deltas of distance and angle
-  float linear = ((float)odom_encoder_right / (float)encoder_cpr * wheel_circumference + (float)odom_encoder_left / (float)encoder_cpr * wheel_circumference) / 2.0;
-//  float angular = ((float)odom_encoder_right / (float)encoder_cpr * wheel_circumference - (float)odom_encoder_left / (float)encoder_cpr * wheel_circumference) / track_width * -1.0;
-  float angular = ((float)odom_encoder_right / (float)encoder_cpr * wheel_circumference - (float)odom_encoder_left / (float)encoder_cpr * wheel_circumference) / track_width;
-#ifdef _ODOM_DEBUG
-/*
-ROS_DEBUG("linear: ");
-ROS_DEBUG(linear);
-ROS_DEBUG(" angular: ");
-ROS_DEBUG(angular);
-ROS_DEBUG("");
-*/
-#endif
+  float linear = (120*wheel_radius*M_PI*(rpm_left_meas + rpm_right_meas)) / 2.0;
+  float angular = (120*wheel_radius*M_PI*(rpm_right_meas - rpm_left_meas)) / track_width;
 
   // Update odometry
-  odom_x += linear * cos(odom_yaw);        // m
-  odom_y += linear * sin(odom_yaw);        // m
+  odom_x += linear * cos(odom_yaw) * dt;        // m
```

Comment:
going back and forth between dt here doesn't really add anything, and currently this `dt` is arguably less accurate than using the fixed 25ms period. I think we should change this back to just updating based on the total incremental distance traveled since the previous message, like it was before.